privacyIDEA-impl/src/main/java/org/privacyidea/action/AbstractChallengeResponseAction.java

```diff
@@ -102,6 +109,47 @@ protected void extractChallengeData(@Nonnull PIResponse piResponse)
         }
     }
 
+    /**
+     * Search for the configured headers in HttpServletRequest and return all found with their values.
+     *
+     * @param request http servlet request
+     * @return headers to forward with their values
+     */
+    protected Map<String, String> getHeadersToForward(HttpServletRequest request)
+    {
+        if (piServerConfigContext.getConfigParams().getForwardHeaders() != null && !piServerConfigContext.getConfigParams().getForwardHeaders().isEmpty())
+        {
+            String cleanHeaders = piServerConfigContext.getConfigParams().getForwardHeaders().replaceAll(" ", "");
+            List<String> headersList = List.of(cleanHeaders.split(","));
+            Map<String, String> headersToForward = new LinkedHashMap<>();

```

Comment:
Put this before the `if (piServerconfigcontext...` and return headersToForward at the end of the function.
Meaning you can change this function easily to have only 1 return statement

Reply:
fixed: [affa5c4](https://github.com/privacyidea/shibboleth-plugin/pull/26/commits/affa5c4906c254c5bc72620f34c8424fcc800410)